ui/components/app/qr-hardware-popover/qr-hardware-popover.js

```diff
@@ -0,0 +1,67 @@
+import React, { useCallback, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getCurrentQRHardwareState } from '../../../selectors';
+import Popover from '../../ui/popover';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  cancelReadQRHardwareCryptoHDKey as cancelReadQRHardwareCryptoHDKeyAction,
+  cancelQRHardwareSignRequest as cancelQRHardwareSignRequestAction,
+  cancelTx,
+} from '../../../store/actions';
+import { clearConfirmTransaction } from '../../../ducks/confirm-transaction/confirm-transaction.duck';
+import QRHardwareWalletImporter from './qr-hardware-wallet-importer';
+import QRHardwareSignRequest from './qr-hardware-sign-request';
+
+const QRHardwarePopover = () => {
+  const t = useI18nContext();
+  const { txData } = useSelector((state) => {
+    return state.confirmTransaction;
+  });
+  const dispatch = useDispatch();
+  const walletImporterCancel = useCallback(
+    () => dispatch(cancelReadQRHardwareCryptoHDKeyAction()),
+    [dispatch],
+  );
+
+  const signRequestCancel = useCallback(() => {
+    dispatch(cancelTx(txData)).then(() => {
+      dispatch(clearConfirmTransaction());
+    });
+    dispatch(cancelQRHardwareSignRequestAction());
+  }, [dispatch, txData]);
+
+  const qrHardware = useSelector(getCurrentQRHardwareState);
+  const { sync, sign } = qrHardware;
+  const showWalletImporter = Boolean(sync && sync.reading);
```

Comment:
I think we can do:  `sync?.reading` on this line, and `sign?.request` on the next.